### ISSUE TITLE
Failed to extract file content from zip files which created from Microsoft Windows Chinese version system

### ISSUE THREAD

**Issue by cedarice** [mattermost/mattermost#25397]
#### Summary
I've many messages like this:
{
  "caller": "app/file.go:803",
  "error": "failed to save the extracted file content: failed to update FileInfo content with id=qwypzjuyofdy3npd77d8as5f6h: pq: invalid byte sequence for encoding \"UTF8\": 0xd6 0xd8",
  "fileInfoId": "qwypzjuyofdy3npd77d8as5f6h",
  "level": "error",
  "msg": "Failed to extract file content",
  "timestamp": "2023-10-08 13:31:11.799 +08:00"
}
According to the database query results, these error files are mostly in zip format and uploaded by users using the Chinese version of Windows. I tested these files using Go language and found that the file names contain gibberish characters (may be GBK encoding reason) after being unzipped. This is likely the reason why the corresponding information cannot be stored in the PostgreSQL database.

#### Steps to reproduce
In the Windows Chinese version system, compress files containing Chinese file names into zip format.

2. After uploading the zip file and sending it out, an error message 'Failed to extract file content' will appear in 'Server Logs'.

3. Based on the file ID mentioned in the error message, query in the postgreSQL database and it happens to be the file that was just uploaded.

#### Expected behavior
I hope to successfully extract the content from these files.

#### Observed behavior (that appears unintentional) 
1. upload the test zip file compressed from windows Chinese version system.
[测试压缩包.zip](https://github.com/mattermost/mattermost/files/13314550/default.zip)

2. Server Logs:
```
{
  "caller": "app/file.go:803",
  "error": "failed to save the extracted file content: failed to update FileInfo content with id=xgx5k67q8b8o9rfkha4b3xy86h: pq: invalid byte sequence for encoding \"UTF8\": 0xd0 0xc2",
  "fileInfoId": "xgx5k67q8b8o9rfkha4b3xy86h",
  "level": "error",
  "msg": "Failed to extract file content",
  "timestamp": "2023-11-10 09:54:27.423 +08:00"
}
```
3. Screenshot of database query results
![image](https://github.com/mattermost/mattermost/assets/12607063/54a2ad59-f166-4ca3-b2de-ae0918a56b09)

#### Possible fixes
Maybe check the file encoding firstly, and then convert to UTF-8?

**Comment by agnivade** [comment]
To correctly encode to utf-8, we need to first know the source encoding. You can do some guesswork by analyzing the file header, but there's no guaranteed way to know the source encoding unless explicitly mentioned by the user. 

Supporting non utf-8 files would require users to enter the encoding scheme while uploading a file so that we can decode it correctly.

**Comment by cedarice** [comment]
I don't quite understand your question. The default encoding for the Chinese version of the Windows operating system is GBK, not UTF-8. For ordinary users who use the Chinese version of windows operating system, they will select some files (the file names of these files contain Chinese characters), pack them into compressed files (zip, rar, 7z, or other), and then send them to others through mattermost successfully. After these files are uploaded, the error log of "Failed to extract file content" will appear in the Server logs of mattermost. I think this is a problem with GBK encoding. In Linux systems or other environments that use UTF-8 encoding by default, when these compressed packages are extracted, the file names of the extracted files are usually garbled.

**Comment by amyblais** [comment]
> I don't quite understand your question. The default encoding for the Chinese version of the Windows operating system is GBK, not UTF-8. For ordinary users who use the Chinese version of windows operating system, they will select some files (the file names of these files contain Chinese characters), pack them into compressed files (zip, rar, 7z, or other), and then send them to others through mattermost successfully. After these files are uploaded, an error log of "Failed to extract file content" will appear in the mattermost server logs.I think this is a problem with GBK encoding. In Linux systems or other environments that use UTF-8 encoding by default, when these compressed packages are extracted, the file names of the extracted files are usually garbled.

cc @agnivade

**Comment by agnivade** [comment]
Yeah I saw it. This is a feature request and needs to be triaged and prioritized by a feature team accordingly.

**Comment by amyblais** [comment]
Thanks, appreciate your feedback @cedarice,

Would you like to [contribute this in the feature idea forum](https://mattermost.uservoice.com/forums/306457-general/) so it can be discussed, upvoted and considered for a [help wanted ticket](https://docs.mattermost.com/process/help-wanted.html)?

Please include a link back to this GitHub issue. If you're interested in implementing, please say so and we'll prioritize the review. 

You get **10** votes in the feature idea forum, and each one influences the future of the project.
